### PR TITLE
Fix batch size reporting

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

**Review by GuanLuo — DISMISSED** [review]

**Review by CoderHam — APPROVED** [review]

File: src/python.cc [hunk 1]
```diff
@@ -355,6 +356,12 @@ ModelInstanceState::~ModelInstanceState()
   }
 }
 
+ModelState*
```
Comment (dzier) [inline src/python.cc:359]: You should return a shared pointer. Is ModelInstance supposed to be a struct or class? If it is a struct, just make model_state a public variable. If not, then don't return raw data. Rather, create methods that can be applied in the object.

Comment (dzier) [reply src/python.cc:359]: For example, you really only need `name` and `max_batch_size`, so create two methods that return just that data, so you are not passing pointers to internal data. That is bad and can be prone to problems down the road.  For example...

```
ModelState * state = instance_state->GetModelState();

.... Do useful work ....

delete state;  // oops, you just hosed the system by accident.
```

**Review by dzier — CHANGES_REQUESTED** [review]

File: src/python.cc (hunk 1 above)
Comment (dzier) [reply src/python.cc:359]: Even more importantly, if you do return a pointer, make it a const pointer unless you intend to alter the data.

File: src/python.cc [hunk 2]
```diff
@@ -118,6 +118,8 @@ class ModelInstanceState : public BackendModelInstance {
       std::vector<TRITONBACKEND_Response*>& responses, size_t r,
       uint32_t& batch_size);
 
+  ModelState* GetModelState();
```
[the remark this answers is not in the document]
Comment (GuanLuo) [reply src/python.cc:121]: You just need to static cast it from `BackendModel` class to your `ModelState` class

File: src/python.cc (hunk 1 above)
Comment (Tabrizian) [reply src/python.cc]: > You should return a shared pointer. Is ModelInstance supposed to be a struct or class? If it is a struct, just make model_state a public variable. If not, then don't return raw data. Rather, create methods that can be applied in the object.

ModelInstanceState is a class. 

I see, thanks for the comment. I ended up deleting that method. It looks like that field was available in the parent class.

File: src/python.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tabrizian) [reply src/python.cc]: Right. I was going through the onnx backend code to see how it's done. It looks like similar changes can be applied there too.

File: src/python.cc [hunk 3]
```diff
@@ -642,12 +640,78 @@ TRITONBACKEND_ModelInstanceExecute(
...
 
-  std::vector<TRITONBACKEND_Response*> responses;
-  responses.reserve(request_count);
-
+  ModelState* model_state =
+      reinterpret_cast<ModelState*>(instance_state->Model());
   uint64_t exec_start_ns = 0;
   SET_TIMESTAMP(exec_start_ns);
 
+  const int max_batch_size = model_state->MaxBatchSize();
```
Comment (dzier) [inline src/python.cc:648]: primitive datatypes do not need to be const.

**Review by dzier — CHANGES_REQUESTED** [review]

File: src/python.cc (hunk 3 above)
Comment (Tabrizian) [reply src/python.cc]: Fixed.

**Review by dzier — APPROVED** [review]

**Pull request merged**